### PR TITLE
Add support for enabling compression on the mezmo exporter

### LINKED ISSUE #32284: Support compression in the mezmo exporter
### Component(s)

exporter/mezmo

### Is your feature request related to a problem? Please describe.

To reduce egress cost it would be nice if the exporter supported compression

### Describe the solution you'd like

A config flag that enables compression

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by recht**
**Description:** <Describe what has changed.>
Add support for enabling compression on the mezmo exporter. This is implemented using a buffer pool to avoid gc overhead, and while I was at it I thought I might as well change the string building to also use the same pool.

**Link to tracking Issue:** 

Fixes #32284

**Testing:**

Unit test added

**Documentation:** 

New config property added to README.md

### CODE REVIEW

**Review by dashpole — APPROVED** [review]
LGTM, but this needs approval from a Mezmo codeowner before merging

File: exporter/mezmoexporter/README.md
Comment (crobert-1) [inline exporter/mezmoexporter/README.md:34]: It would be good to add a note here that it's disabled by default, as well as a list of valid values, or an example value that could show what is expected here.

File: exporter/mezmoexporter/config.go
Comment (recht) [reply exporter/mezmoexporter/config.go:47]: I couldn't really see any reason for making it configurable, given that the exporter is for a single specific service, so the compression method seems to be an internal detail... but I can definitely change it if that's better.

**Pull request closed without merging**
